### PR TITLE
Bugfix for ocean submesoscale feature

### LINKED ISSUE #5459: Invalid values when submesoscale variables are masked
Fill value masking of the submesoscale variable `normalMLEVelocity` resulted in bad values in `vertMLEBolusVelocityTop` (Inf) that affected `timeMonthly_avg_mocStreamvalLatAndDepth`. The issue arises when (a) the masking array for `normalMLEVelocity` is specified in the Registry and (b) the `normalMLEVelocity` variable is included in a timeSeriesStats stream (regardless of whether masking is turned on via `useMissingValMask`). In MPAS-Ocean standalone, this bug manifests as state validation errors.

### PR DESCRIPTION
**Pull request by cbegeman**
Fill value masking of the submesoscale variable `normalMLEVelocity` resulted in bad values in `vertMLEBolusVelocityTop` that affected `timeMonthly_avg_mocStreamvalLatAndDepth`. This bugfix removes masking of `normalMLEVelocity`. 

Fixes #5459

[BFB]

### CODE REVIEW

**Comment by cbegeman** [comment]
### Testing

This PR is BFB with master for a variant of the QU240 `daily_output_test` on chrysalis (intel, openmpi) where the submesoscale eddy parameterization is turned on and `normalMLEVelocity` and `vertMLEBolusVelocityTop` are added to the timeDailyStats stream. For the master branch, these variables cannot be included in this stream without introducing bad values.

**Comment by xylar** [comment]
@cbegeman, we'll also need to make an issue reporting the bug this fixes, and indicate here that this PR closes that issue.  I can help with that tomorrow.

**Comment by jonbob** [comment]
@cbegeman -- I think we will need a similar PR into the maint-2.1 branch

**Comment by cbegeman** [comment]
@jonbob Let me know if you'd like me to open that PR now or after you've done further E3SM testing.

**Comment by rljacob** [comment]
This just fixes some output?  It doesn't change answers?

**Comment by jonbob** [comment]
@rljacob: It doesn't change E3SM state answers, but we get bad values in the ocean output we use for mpas-analysis. So we can't really analyze results from runs using master or v2.1

**Comment by cbegeman** [comment]
The equivalent PR to maint-2.1 is here https://github.com/E3SM-Project/E3SM/pull/5461

**Comment by darincomeau** [comment]
Looks reasonable to me and I'm happy to approve based on developer testing, unless you'd like me to do some other testing?

**Review by xylar — APPROVED** [review]
This worked for me in compass.  I ran all the QU240 tests with the submesoscale parameterizaiton turned on.  Using the compass E3SM-Project submodule, 8 forward tests failed including the daily and monthly output tests. With this fix, all tests passed.

**Comment by xylar** [comment]
@jonbob, could you run a month (or more) of an EC30to60 test and point me to the output so we can both sanity check it?

**Review by mark-petersen — APPROVED** [review]
Yes, I agree with this change. `normalMLEvelocity` variable format should match `normalVelocity`.

**Comment by jonbob** [comment]
@xylar -- I made two two-month runs, a baseline and a test of this branch. The results can be found at:

/lcrc/group/acme/ac.jwolfe/scratch/chrys/20230221.5458base.chrys/run
/lcrc/group/acme/ac.jwolfe/scratch/chrys/20230221.5458test.chrys/run

**Comment by xylar** [comment]
Thanks @jonbob. Taking a look now.

**Comment by xylar** [comment]
I wrote a little script to test the `timeSeriesStatsMonthly` output of the two test runs:
``` python
#!/usr/bin/env python
import xarray as xr
import numpy as np


#filename = '/lcrc/group/acme/ac.jwolfe/scratch/chrys/20230221.5458base.chrys/run/20230221.5458base.chrys.mpaso.hist.am.timeSeriesStatsMonthly.0001-02-01.nc'
filename = '/lcrc/group/acme/ac.jwolfe/scratch/chrys/20230221.5458test.chrys/run/20230221.5458test.chrys.mpaso.hist.am.timeSeriesStatsMonthly.0001-02-01.nc'

ds = xr.open_dataset(filename, mask_and_scale=False)

all_finite = True
for var in ds.data_vars:
    data = ds[var].values
    if data.dtype not in [np.float32, np.float64, np.int32, np.int64]:
        continue
    finite = np.all(np.isfinite(data))
    if not finite:
        all_finite = False
        print('!!!! bad news !!!!')
    print(f'{var}: {finite}')

print(all_finite)
```
From the `base` run, I see that there are non-finite values in:
```
timeMonthly_avg_vertMLEBolusVelocityTop
timeMonthly_avg_mocStreamvalLatAndDepthMLE
timeMonthly_avg_mocStreamvalLatAndDepthRegionMLE
timeMonthly_avg_mocStreamvalLatAndDepth
timeMonthly_avg_mocStreamvalLatAndDepthRegion
```
so a couple more fields than we previously had on our radar but all related to submesoscale.

With the new run, everything is finite.  So I think we're good.

**Comment by xylar** [comment]
@darincomeau, I don't think we need a review from you at this point, I think we're good.  Thanks for taking a look.

**Comment by jonbob** [comment]
passes:
* ERS_Ld5.T62_oQU120.CMPASO-NYF.chrysalis_intel
* SMS_D_Ld3.T62_oQU120.CMPASO-IAF.chrysalis_intel
* ERS.ne11_oQU240.WCYCL1850NS.chrysalis_intel
* SMS_D_Ld1.ne30pg2_EC30to60E2r2.WCYCL1850.chrysalis_intel.allactive-wcprod

merged to next

**Comment by jonbob** [comment]
merged to master

**Comment by cbegeman** [comment]
@xylar and @jonbob Thank you for all the testing you did for this PR!

**Pull request merged**
